torch/csrc/jit/interpreter.cpp

```diff
@@ -981,6 +1017,8 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
             ++af.pc;
           } break;
           case TAIL_CALL: {
+            // build a bailout grpah
+            af.functions[inst.X]->ensure_defined();
```

Comment:
This seems fragile, can we just make `get_executor` call `ensured_defined`?

Reply:
Also, I just realized this is the first place where we introduce a possible race condition. Function needs to hold its lock (there is one for the executor already) in order to run its function creator.

Reply:
> This seems fragile, can we just make get_executor call ensured_defined?
moved `ensure_defined` to `get_executor`.

> Also, I just realized this is the first place where we introduce a possible race condition. Function needs to hold its lock (there is one for the executor already) in order to run its function creator.

hhmm, I'll bug you about this.